Landing Page

TheAmanM opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!

**NOTE**:

- It's a good idea to open an issue first to discuss potential changes.
- Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

-->

# Landing Page [(View here)](https://auth-docs-git-fork-theamanm-feat-landing-page-authjs.vercel.app/)

<img width="1908" height="940" alt="image" src="https://github.com/user-attachments/assets/2b4ade97-e87e-459a-99e1-9f3240ef7a59" />


Auth JS is awesome. The landing page sucks. Hundreds of hours, if not more have gone into creating a free and open source way for developers to have an easier time integrating authentication into their applications, whether they're a Next JS developer or making an Express server.

## Solution

This PR is designed to re-design and implement a new landing page. It also refactors `index.mdx`.

<!-- What changes are being made? What feature/bug is being fixed here? -->

## 🎫 Affected issues

<!--
Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR. And include text like the following to close them automatically when this is merged:
-->

Addresses #13194

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

TheAmanM committed:
hero section and social proof

TheAmanM committed:
styled rest of the page

TheAmanM committed:
dark mode compatibility

TheAmanM committed:
fixed testimonials

TheAmanM committed:
fixed features

TheAmanM committed:
fixed logos

TheAmanM committed:
hover effect on provider icons

TheAmanM committed:
fix: links

TheAmanM committed:
responsive cta

TheAmanM committed:
fixed hero image

TheAmanM committed:
fix: code hero section

TheAmanM committed:
fixed statistics

TheAmanM committed:
responsive provider logos

TheAmanM committed:
fix: github icon

TheAmanM committed:
hero code theming

TheAmanM committed:
minor text and theming fixes

TheAmanM committed:
refactoring

halvaradop commented on docs/components/Providers/Marquee.tsx:
```diff
@@ -0,0 +1,212 @@
+    />
+    <div className="pointer-events-none absolute -top-10 left-1/2 -translate-x-1/2 scale-0 opacity-0 transition-all duration-200 group-hover:scale-100 group-hover:opacity-100">
+      <div className="whitespace-nowrap rounded-md border border-neutral-700 bg-neutral-900 px-2 py-1 text-xs text-neutral-200">
+        {name}.svg
```
I believe it’s better to remove the `.svg` extension and use only the provider name as it's registered in the `manifest.json` file.

halvaradop commented on docs/components/Providers/Marquee.tsx:
```diff
@@ -0,0 +1,212 @@
+        </div>
+      </div>
+
+      <style>
+        {`
+          @keyframes scroll-left {
+            0% { transform: translateX(0); }
+            100% { transform: translateX(-50%); }
+          }
+          .animate-scroll-left {
+            animation: scroll-left 15s linear infinite;
+          }
+          
+          @keyframes scroll-right {
+            0% { transform: translateX(-50%); }
+            100% { transform: translateX(0); }
+          }
+          .animate-scroll-right {
+            animation: scroll-right 15s linear infinite;
+          }
+
+          @media (min-width: 1024px) {
+            .animate-scroll-left {
+              animation-duration: 75s;
+            }
+            .animate-scroll-right {
+              animation-duration: 75s;
+            }
+          }
+
+          .filter-white {
+            filter: brightness(0) invert(1);
+          }
+        `}
+      </style>
```
I recommend removing the `<style>` HTML tag and moving its contents to the [globals.css](https://github.com/nextauthjs/next-auth/blob/main/docs/pages/global.css) file. This helps maintain a centralized entry point for CSS styles and keeps the codebase cleaner and more consistent.

halvaradop replied:
I recommend removing the `<style>` HTML tag and moving its contents to the [globals.css](https://github.com/nextauthjs/next-auth/blob/main/docs/pages/global.css) file. This helps maintain a centralized entry point for CSS styles and keeps the codebase cleaner and more consistent.

halvaradop reviewed (commented):
Great work on this! Here are a few suggestions to consider for improvement.

halvaradop commented on docs/components/Hero/CodeTabs.tsx:
```diff
@@ -0,0 +1,154 @@
+import qwikIcon from "../../public/img/etc/qwik.svg"
+import sveltekitIcon from "../../public/img/etc/sveltekit.svg"
+
+const ExpressIcon = () => (
+  <img src={expressIcon.src} alt="Express" className="h-6 w-6 dark:invert" />
+)
+const NextIcon = () => (
+  <img
+    src={nextIcon.src}
+    alt="Next.js"
+    className="h-6 w-6 filter dark:invert"
+  />
+)
+const QwikIcon = () => <img src={qwikIcon.src} alt="Qwik" className="h-6 w-6" />
+const SvelteKitIcon = () => (
+  <img src={sveltekitIcon.src} alt="SvelteKit" className="h-6 w-6" />
+)
```
Just a reminder: the site runs on `Next.js`, so it's best to use the `Image` component from `next/image` for any images. It handles optimization and responsiveness out of the box. [Docs](https://nextjs.org/docs/14/pages/api-reference/components/image)

halvaradop commented on docs/components/Hero/CodeTabs.tsx:
```diff
@@ -0,0 +1,154 @@
+)
+const CaretRight = () => <span>›</span>
+
+export const CodeTabs = () => {
```
I think you can reuse the `RichTabs` component from the previous version and simply update the CSS styles and code highlighting logic.

halvaradop commented on docs/components/Providers/Marquee.tsx:
```diff
@@ -0,0 +1,212 @@
+import React from "react"
+
+const providers = [
```
There's no need to manually create an array of providers—you can use the JSON file that lists them when they're added to auth.js.
Here's the reference: https://github.com/nextauthjs/next-auth/blob/39dd3b92de194c1a835f2d87631f4deb9d9fdf65/docs/pages/data/manifest.json#L45

halvaradop commented on docs/components/Providers/Marquee.tsx:
```diff
@@ -0,0 +1,212 @@
+      }}
+    />
+    <div className="pointer-events-none absolute -top-10 left-1/2 -translate-x-1/2 scale-0 opacity-0 transition-all duration-200 group-hover:scale-100 group-hover:opacity-100">
+      <div className="whitespace-nowrap rounded-md border border-neutral-700 bg-neutral-900 px-2 py-1 text-xs text-neutral-200">
+        {name}.svg
+      </div>
```
I believe it’s better to remove the `.svg` extension and use only the provider name as it's registered in the `manifest.json` file.

halvaradop reviewed (commented):
Great work on this! Here are a few suggestions to consider for improvement.

TheAmanM committed:
migrated <img> to <Image>

TheAmanM committed:
switched providers to manifest.json => reduced redundancy

TheAmanM committed:
moved styles to globals.css

TheAmanM committed:
fix: migrated to RichTabs

TheAmanM commented:
Thank you for the direction - it's definitely not easy to navigate a new codebase! 

I've made the desired changes:

- Swapped <img /> to <Image />
- Switched from a custom providers array to reusing providersOAuth
- Changes `{name}.svg` to just the name as registered in manifest.json
- Moved styles to globals.css
- Reused RichTabs and implemented CSS

halvaradop commented:
Thanks for taking my suggestions into account—you’ve done a great job!

TheAmanM committed:
Updated index.tsx to remove asterisks

TheAmanM commented:
> Thanks for taking my suggestions into account. You’ve done a great job!
> 
> 
> 

Thank you so much! This is my first time - contributing to OSS at this scale, and I'm glad I took the initiative!

TheAmanM committed:
fix: text sizing
